### PR TITLE
[12.0] [MIG] [WIP] Migration of 'quality_control_mrp'

### PR DESCRIPTION
**Pull request by NachoAlesLopez**
@guadaltech

- [ ] Fix Transbot commits.
- [X] Check if tests passes.

### CODE REVIEW

File: quality_control_mrp/tests/test_quality_control_mrp.py [hunk 1]
```diff
@@ -38,11 +38,15 @@ def test_inspection_create_for_product(self):
                 'test': self.test.id,
             }
         )]
+        lot = self.env['stock.production.lot'].create({
```
Comment (Tardo) [inline quality_control_mrp/tests/test_quality_control_mrp.py]: You can avoid "lot/serial number" usage setting the product tracking to 'none'.

File: quality_control_mrp/models/qc_inspection.py [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+    def get_production(self):
+        for inspection in self:
+            if inspection.object_id:
+                if inspection.object_id._name == 'stock.move':
+                    inspection.production_id = inspection.object_id.production_id
+                elif inspection.object_id._name == 'mrp.production':
+                    inspection.production_id = inspection.object_id
+
+    @api.depends('object_id')
+    def _get_product(self):
```
Comment (Tardo) [inline quality_control_mrp/models/qc_inspection.py]:
```suggestion
    def _compute_product_id(self):
```

File: quality_control_mrp/models/qc_inspection.py [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+        """Overriden for getting the product from a manufacturing order."""
+        for inspection in self:
+            super(QcInspection, inspection)._get_product()
+            if inspection.object_id and\
+                    inspection.object_id._name == 'mrp.production':
+                inspection.product = inspection.object_id.product_id
```
Comment (Tardo) [inline quality_control_mrp/models/qc_inspection.py]:
```suggestion
                inspection.product_id = inspection.object_id.product_id
```

File: quality_control_mrp/tests/__init__.py [hunk 4]
```diff
@@ -0,0 +1,4 @@
+# -*- coding: utf-8 -*-
```
Comment (Tardo) [inline quality_control_mrp/tests/__init__.py]: :fire:

File: quality_control_mrp/tests/test_quality_control_mrp.py [hunk 5]
```diff
@@ -0,0 +1,166 @@
+# -*- coding: utf-8 -*-
```
Comment (Tardo) [inline quality_control_mrp/tests/test_quality_control_mrp.py]: :fire:

**Review by Tardo — CHANGES_REQUESTED** [review]
Thanks for the work! Please, add a migration script to rename columns.

**Comment by NachoAlesLopez** [comment]
@Tardo Thanks for the comments! Most of them should be fixed now, I'm only missing the migration scripts. It is going to take me a bit, since I don't have experience with migration scripts.

**Comment by NachoAlesLopez** [comment]
@Tardo In order to make the migration to 12.0, I will need to migrate this module to 11.0 first. I've started the PR in #443. It's basically a backport of this PR but with the migration script. Do you mind checking it out?

**Comment by pedrobaeza** [comment]
You don't need to migrate the module to 11.0. You can do directly the migration from 10.0 to 12.0. Only change branches in the commands.

**Comment by NachoAlesLopez** [comment]
@pedrobaeza Thanks for the information! I'll try the migration script I've made in the 11.0 PR and push it here if it works when migrating from 10.0.

File: quality_control_mrp/migrations/10.0.1.0.0/pre-migration.py [hunk 6]
```diff
@@ -0,0 +1,13 @@
+from openupgradelib import openupgrade
```
Comment (pedrobaeza) [inline quality_control_mrp/migrations/10.0.1.0.0/pre-migration.py]: Incorrect version folder: it must be `12.0.1.0.0`

File: quality_control_mrp/migrations/10.0.1.0.0/pre-migration.py [hunk 7]
```diff
@@ -0,0 +1,13 @@
...
+
+field_renames = [
+    ("mrp.production", "mrp_production", "qc_inspections",
+     "qc_inspections_ids"),
```
Comment (pedrobaeza) [inline quality_control_mrp/migrations/10.0.1.0.0/pre-migration.py]: As you are renaming, please rename to proper name: `qc_inspection_ids`

**Comment by elvise** [comment]
Hi, any good news for merge...?
If I can help please let me know...

File: quality_control_mrp/readme/CONTRIBUTORS.rst [hunk 8]
```diff
@@ -0,0 +1,3 @@
+* Pedro M. Baeza <pedro.baeza@serviciosbaeza.com>
+* Oihane Crucelaegui <oihanecrucelaegi@avanzosc.es>
+* Simone Rubino <simone.rubino@agilebg.com>
```
Comment (ps-tubtim) [inline quality_control_mrp/readme/CONTRIBUTORS.rst:3]: Please add you as a Contributor.

**Review by ps-tubtim — COMMENTED** [review]
Functional test: OK.

**Review by Tardo — APPROVED** [review]
:+1: Code review

**Comment by pedrobaeza** [comment]
Let's merge it as there are no blocking points, although it would be great to would have the field renaming to the good one.

/ocabot merge

**Pull request merged**